probe/endpoint/connection_tracker.go

```diff
@@ -0,0 +1,260 @@
+	hostNodeID := report.MakeHostNodeID(t.conf.HostID)
+
+	if t.ebpfTracker != nil {
+		t.performEbpfTrack(rpt, hostNodeID)
+		return
+	}
+
+	// seenTuples contains information about connections seen by conntrack and it will be passed to the /proc parser
+	seenTuples := map[string]fourTuple{}
+	if t.flowWalker != nil {
+		t.performFlowWalk(rpt, &seenTuples)
+	}
+	if t.conf.WalkProc {
+		t.performWalkProc(rpt, hostNodeID, &seenTuples)
+	}
+}
+
+func (t *connectionTracker) performFlowWalk(rpt *report.Report, seenTuples *map[string]fourTuple) {
+	// Consult the flowWalker for short-lived connections
+	extraNodeInfo := map[string]string{
+		Conntracked: "true",
+	}
+	t.flowWalker.walkFlows(func(f flow, alive bool) {
+		tuple := fourTuple{
+			f.Original.Layer3.SrcIP,
+			f.Original.Layer3.DstIP,
+			uint16(f.Original.Layer4.SrcPort),
+			uint16(f.Original.Layer4.DstPort),
+		}
+		// Handle DNAT-ed short-lived connections.
+		// The NAT mapper won't help since it only runs periodically,
+		// missing the short-lived connections.
+		if f.Original.Layer3.DstIP != f.Reply.Layer3.SrcIP {
+			tuple = fourTuple{
+				f.Reply.Layer3.DstIP,
+				f.Reply.Layer3.SrcIP,
+				uint16(f.Reply.Layer4.DstPort),
+				uint16(f.Reply.Layer4.SrcPort),
+			}
+		}
+
+		(*seenTuples)[tuple.key()] = tuple
+		t.addConnection(rpt, tuple, "", extraNodeInfo, extraNodeInfo)
+	})
+}
+
+func (t *connectionTracker) performWalkProc(rpt *report.Report, hostNodeID string, seenTuples *map[string]fourTuple) error {
+	conns, err := t.conf.Scanner.Connections(t.conf.SpyProcs)
+	if err != nil {
+		return err
+	}
+	for conn := conns.Next(); conn != nil; conn = conns.Next() {
+		var (
+			namespaceID string
+			tuple       = fourTuple{
+				conn.LocalAddress.String(),
+				conn.RemoteAddress.String(),
+				conn.LocalPort,
+				conn.RemotePort,
+			}
+			toNodeInfo   = map[string]string{Procspied: "true"}
+			fromNodeInfo = map[string]string{Procspied: "true"}
+		)
+		if conn.Proc.PID > 0 {
+			fromNodeInfo[process.PID] = strconv.FormatUint(uint64(conn.Proc.PID), 10)
+			fromNodeInfo[report.HostNodeID] = hostNodeID
+		}
+
+		if conn.Proc.NetNamespaceID > 0 {
+			namespaceID = strconv.FormatUint(conn.Proc.NetNamespaceID, 10)
+		}
+
+		// If we've already seen this connection, we should know the direction
+		// (or have already figured it out), so we normalize and use the
+		// canonical direction. Otherwise, we can use a port-heuristic to guess
+		// the direction.
+		canonical, ok := (*seenTuples)[tuple.key()]
+		if (ok && canonical != tuple) || (!ok && tuple.fromPort < tuple.toPort) {
+			tuple.reverse()
+			toNodeInfo, fromNodeInfo = fromNodeInfo, toNodeInfo
+		}
+		t.addConnection(rpt, tuple, namespaceID, fromNodeInfo, toNodeInfo)
+	}
+	return nil
+}
+
+func (t *connectionTracker) performEbpfTrack(rpt *report.Report, hostNodeID string) error {
+	t.ebpfTracker.walkConnections(func(e ebpfConnection) {
+		fromNodeInfo := map[string]string{
+			Procspied: "true",
+			EBPF:      "true",
```

Comment:
Have you checked the implications of these tags on the UI side?

Why is Procspied still marked? Backwards compatibility? (If so please add a comment)